### PR TITLE
Feature/43 add beginner and extreme levels

### PR DESCRIPTION
**Pull request by en-chang**
- Added beginner and extreme difficulty options.
- Being honest I added code to the src/models for both the client and server but I don't know what those actually do so if you could explain that would be cool.
add beginner and extreme levels

### CODE REVIEW

File: src/components/DifficultySwitcher/DifficultySwitcher.tsx [hunk 1]
```diff
@@ -23,6 +23,15 @@ const DifficultSwitcher: FC<DifficultSwitcherProps> = ({
 }) => {
   return (
     <div className={classnames("difficulty-switcher", className)}>
+      <button
```
Comment (bkchu) [inline src/components/DifficultySwitcher/DifficultySwitcher.tsx]: Now that there are 5 difficulty buttons here, I wonder if there's a way here to do this programmatically.

File: functions/get-sudoku-board.js [hunk 2]
```diff
@@ -3,9 +3,11 @@ const _ = require("lodash")
 exports.handler = async function (event, context, callback) {
   const { difficulty } = event.queryStringParameters
   const difficulties = {
+    beginner: 71,
```
Comment (bkchu) [inline functions/get-sudoku-board.js]: Did you read the comments in the ticket before you started this story?

**Review by bkchu — DISMISSED** [review]

File: src/components/DifficultySwitcher/DifficultySwitcher.tsx [hunk 3]
```diff
@@ -21,35 +21,37 @@ const DifficultSwitcher: FC<DifficultSwitcherProps> = ({
...
+    "EXTREME"
+  ]
+  const difficultyLevels = [
+    Difficulty.BEGINNER,
+    Difficulty.EASY,
+    Difficulty.MEDIUM,
+    Difficulty.HARD,
+    Difficulty.EXTREME
+  ]
+  var difficultyButtons = []
```
Comment (bkchu) [inline src/components/DifficultySwitcher/DifficultySwitcher.tsx]: Nice. Try refactoring with Array.map()

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/map

File: src/components/DifficultySwitcher/DifficultySwitcher.tsx [hunk 4]
```diff
@@ -21,35 +21,37 @@ const DifficultSwitcher: FC<DifficultSwitcherProps> = ({
...
-        className={classnames("difficulty-switcher__button", {
-          "difficulty-switcher__button--selected":
-            difficulty === Difficulty.EASY,
-        })}
-        onClick={() => setDifficulty(Difficulty.EASY)}
-      >
-        EASY
-      </button>
-      <button
+  const levels = [
```
Comment (bkchu) [inline src/components/DifficultySwitcher/DifficultySwitcher.tsx]: Are these being used anywhere? O-o

Comment (en-chang) [reply src/components/DifficultySwitcher/DifficultySwitcher.tsx]: I don't know how to quote my own code but it's used on line 48.

File: src/components/DifficultySwitcher/DifficultySwitcher.tsx (hunk 3 above)
Comment (en-chang) [reply src/components/DifficultySwitcher/DifficultySwitcher.tsx]: Alrighty

File: src/components/DifficultySwitcher/DifficultySwitcher.tsx [hunk 5]
```diff
@@ -21,35 +21,27 @@ const DifficultSwitcher: FC<DifficultSwitcherProps> = ({
...
   className
 }) => {
+  const difficultyLevels = [
+    Difficulty.BEGINNER,
+    Difficulty.EASY,
+    Difficulty.MEDIUM,
+    Difficulty.HARD,
+    Difficulty.EXTREME
+  ]
+  var difficultyButtons = difficultyLevels.map(level =>
```
Comment (bkchu) [inline src/components/DifficultySwitcher/DifficultySwitcher.tsx]: change this to `const` and I think it should be good.

**Review by bkchu — CHANGES_REQUESTED** [review]

File: src/components/DifficultySwitcher/DifficultySwitcher.tsx (hunk 5 above)
Comment (en-chang) [reply src/components/DifficultySwitcher/DifficultySwitcher.tsx]: gotcha

**Review by bkchu — APPROVED** [review]

**Pull request merged**